Magento_Theme/styles/modules/_date-picker-trigger.scss

```diff
@@ -0,0 +1,12 @@
+.ui-datepicker-trigger {
+    @extend .button;
+    @extend .button--icon;
+    background-image: url("../images/icons/calendar.svg") !important;
+    background-size: 25px 25px !important;
+    background-repeat: no-repeat !important;
+    background-position: center center !important;
+
+    span {
+        display: none;
+    }
+}
```

Comment:
it's not the ultimate solution but the init script for the date-picker widget is hardcoded in 
vendor/magento/framework/View/Element/Html/Date.php